### DISCUSSION TITLE
Role management and give access to specific role in an activity

### DISCUSSION THREAD

**Discussion by rezabashiri** [elsa-workflows/elsa-core#475] [Q&A]
Think about a scenario in document approval workflow  a document should be reviewed by manager and at next stage it should be reviewed by another person  .

How to implement such scenario ?

**Comment by sfmskywalker** [comment]
You could implement this with a couple of custom activities. One such activity might be an `AwaitApproval` activity that blocks the workflow until a signal is received from some user with the appropriate role. The `AwaitApproval` activity would be configurable by having a `RequiredRole` property or even a specific `RequiredUser` property that the workflow user would set to the user role or specific user that is required to approve the document.

When the user approves the document, the workflow continues to the next `AwaitApproval` activity, this time waiting for approval from a user from a different role or different specific user.
